app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_by_comment_and_curiosity_by_user_scope, only: [:update, :destroy]
+  before_action :find_curiosity_card, only: [:edit, :create]
+
+  def create
+    @comment = @curiosity.comments.new(comment_params)
+    if @comment.save
+      redirect_to @curiosity
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @comment = Comment.find(params[:id])
+  end
+
+  def update
+    @comment&.update(comment_params)
+
+    redirect_to curiosity_card_path(@curiosity)
+  end
+
+  def destroy
+    @comment&.destroy
+
+    redirect_to curiosity_card_path(@curiosity), status: :see_other
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:message).merge(user_id: current_user.id)
+  end
+
+  def find_by_comment_and_curiosity_by_user_scope
```

Comment:
Based on what method it's doing, I think name should be something more like `find_comment_and_curiosity_by_user`

Reply:
Thanks, I think this is a better name, I'll correct it :D